### ISSUE TITLE
Pagination?

### ISSUE THREAD

**Issue by zaptrem** [flyerhq/react-native-firebase-chat-core#19]
Hello, I'm interested in migrating to this from Gifted-Chat. How do you handle long chat histories? Do you just load every document whenever the chat is opened or do you implement some type of pagination? If so, does it work with live updates?

Thanks!

**Comment by demchenkoalex** [comment]
Hey @zaptrem - I believe gifted chat is a UI implementation only? we have 2 different projects, one is a chat UI and second one firebase implementation. Right now I am more focused on Flutter counterparts (I plan to update RNs chat UI to Flutters level soon) but firebase is more suitable for small or PoC projects and is not properly optimised to work with pagination as well.

**Comment by zaptrem** [comment]
> Hey @zaptrem - I believe gifted chat is a UI implementation only? we have 2 different projects, one is a chat UI and second one firebase implementation. Right now I am more focused on Flutter counterparts (I plan to update RNs chat UI to Flutters level soon) but firebase is more suitable for small or PoC projects and is not properly optimised to work with pagination as well.

Thanks! Are there plans to add pagination to the Firestore implementation (or potentially switch to RTDB)? If not I'll just roll my own with your UI once it's ready.

Either way, thanks!

**Comment by demchenkoalex** [comment]
Yeah, it will definitely be faster if you quickly migrate to RTDB yourself, cause you know, I have like 10 libraries to maintain and a full-time job, so it will not happen in the near future I suppose, only if I will have this feeling that I really want to do it, but it happens rarely :D
